src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -637,8 +650,12 @@ internal void SendMessage(Message message, ActivationData sendingActivation = nu
             // second, we check for a strategy associated with the target's interface. third, we check for a strategy associated with the activation sending the
             // message.
             var strategy = targetAddress.Grain.IsGrain ? catalog.GetGrainPlacementStrategy(targetAddress.Grain) : null;
+            var request = message.IsUsingIfaceVersion
+                ? message.GetDeserializedBody(this.serializationManager) as InvokeMethodRequest
+                : null;
+            var target = new PlacementTarget(message.TargetGrain, request?.InterfaceId ?? 0, request?.InterfaceVersion ?? 0);
```

Comment:
Not sure here if it would be preferable to use a `struct` instead. Would be the compiler clever enough to "inline" that?

But I definitely don't like to have function with too many parameters